### ISSUE TITLE
SyntaxError: Invalid regular expression: /c++/: Nothing to repeat

### ISSUE THREAD

**Issue by ArenoRobert** [sergiodlopes/jquery-flexdatalist#188]
Hi,

I have a problem when searching for C++ using the following config. I've been trying to manipulate the search words in many different ways but I don't manage to handle it.
Any suggestions?

`	            $('#list').flexdatalist({
	                url: '/app/api/filter/skills/',
	                selectionRequired : true,
	                minLength: 1,
	                focusFirstResult: true,
	                searchContain: true,
	                searchIn: 'title',
	                noResultsText: '"{keyword}" hittades inte',
	                valueProperty: 'id',
	                maxShownResults: 10
	            });`

`uncaught SyntaxError: Invalid regular expression: /c++/: Nothing to repeat
    at new RegExp (<anonymous>)
    at Object.highlight (jquery.flexdatalist.min.js:formatted:749)
    at Object.matches (jquery.flexdatalist.min.js:formatted:741)
    at Object.get (jquery.flexdatalist.min.js:formatted:719)`

**Comment by griffith-icp** [comment]
Did you find a fix to this issue

**Comment by ArenoRobert** [comment]
No I didn’t unfortunately. Do you have any ideas?

**Comment by griffith-icp** [comment]
Yes, I found a way out. We just had to escape the string before converting it to regex.
You realize `+` is a valid regex operator.

```
function escapeRegExp (string) {
    return string.replace(/[.*+?^${}()|[\]\\]/g, '\\$&'); // $& means the whole matched string
  }
```

**Comment by ArenoRobert** [comment]
That's great! Did you modify the orginal code or where you able to inject your code in a method/option? I was looking at the method "normalizeString", but that is too late I guess?

**Comment by griffith-icp** [comment]
I wasn't really using this particular library.
I am building a react application and I happened to face this same issue, a random search brought me here. But try escaping the string before passing it as a payload to the jquery plugin.
